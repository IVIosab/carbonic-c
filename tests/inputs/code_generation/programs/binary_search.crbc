routine main () : integer is 
    var target : integer is 135;
    print(target);
    var steps : integer is 0;
    var left : integer is 0;
    print(left);
    var right : integer is 100000;
    print(right);

    while left < right loop
        steps := steps + 1;
        var middle : integer is (left + (right-left)/2);
        if middle > target then
            right := middle;
        else 
            left := middle+1;
        end
    end
    print(steps);
    return 0;
end
