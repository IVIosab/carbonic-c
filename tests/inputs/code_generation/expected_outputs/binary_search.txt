135
0
100000
17